golang/thrift/outbound.go

```diff
@@ -0,0 +1,323 @@
+package thrift
+
+import "bytes"
+import "errors"
+import "github.com/apache/thrift/lib/go/thrift"
+import tchannel "github.com/uber/tchannel/golang"
+import "golang.org/x/net/context"
+import "time"
+
+// NewTChannelOutboundProtocol creates a TChannelOutboundProtocol
+func NewTChannelOutboundProtocol(remoteHostPort, remoteServiceName, remoteProcessorName string) (*TChannelOutboundProtocol, error) {
+	tchannel, err := tchannel.NewChannel("0.0.0.0:0", nil)
```

Comment:
The context also typically represents a goroutine level object, so I think it's fine to pass it into the TChannelOutboundProtocol and share it between calls.  In server applications, the context is created when an incoming request is received, and then is shared by all calls made during the processing of that request.  In pure clients (e.g. command line utilities), you usually create the context once at the start of main() and then use it for every subsequent invocation.